packages/library/src/send-transaction-internal.ts

```diff
@@ -23,7 +28,7 @@ interface SendAndConfirmDurableNonceTransactionConfig
             'getNonceInvalidationPromise' | 'getRecentSignatureConfirmationPromise'
         >,
     ) => Promise<void>;
-    transaction: IDurableNonceTransaction & SendableTransaction;
+    transaction: FullySignedTransaction & { lifetimeConstraint: TransactionDurableNonceLifetime };
```

Comment:
Maybe `Readonly<{...}>` all of these?

Reply:
Good idea! I'll do this in an update to https://github.com/solana-labs/solana-web3.js/pull/2486 to avoid rebase conflicts there